### DISCUSSION TITLE
Is there a way to apply "x-go-type-skip-optional-pointer": true to the entire oapi code generation process?

### DISCUSSION THREAD

**Discussion by Tanz0rz** [oapi-codegen/oapi-codegen#1460] [Q&A]
I do not have the ability to edit the swagger files. What is the best way to achieve this behavior?

**Comment by jamietanna** [comment]
You may be able to use Nullable Types that were introduced in https://github.com/deepmap/oapi-codegen/releases/tag/v2.1.0 but it depends on what your use-case is for this?

Otherwise, we could add a global configuration for it - PRs welcome!

**Reply by deitch** [reply]
I don't mind taking a stab at this, as it would be useful for me as well.

It wasn't hard to figure out where to plumb in the option, [in Configuration](https://github.com/deepmap/oapi-codegen/blob/da9077bb5ffecff9185dc13d23760580d899333d/pkg/codegen/configuration.go#L14), specifically [OutputOptions](https://github.com/deepmap/oapi-codegen/blob/da9077bb5ffecff9185dc13d23760580d899333d/pkg/codegen/configuration.go#L108-L142).

And the tag is defined as `extPropGoTypeSkipOptionalPointer ` [here](https://github.com/deepmap/oapi-codegen/blob/da9077bb5ffecff9185dc13d23760580d899333d/pkg/codegen/extension.go#L12). That `const` is referenced in 2 places:

* inside `GenerateGoSchema()` [here](https://github.com/deepmap/oapi-codegen/blob/master/pkg/codegen/schema.go#L295)
* inside `GenFieldsFromProperties()` [here](https://github.com/deepmap/oapi-codegen/blob/master/pkg/codegen/schema.go#L295)

Those set the Schema based on the values of various things, like the tags.

The right approach seems to be to check the default from the config, and, if it is set _or_ if it is in the tag, then set it as true.

The problem is I don't see how the global properties are plumbed through to where those calls are made.

If you can point me in the right direction @jamietanna , I can open a PR on it.

**Reply by deitch** [reply]
Any interest?

**Reply by deitch** [reply]
Bump

**Comment by aksdb** [comment]
I added this feature in a fork I now opened as PR #1694, where I hope we can discuss what's missing to get this upstream.
